badgecheck/tasks/graph.py

```diff
@@ -21,11 +23,22 @@ def fetch_http_node(state, task_meta):
             return task_result(message='Successfully fetched image from {}'.format(url))
         return task_result(success=False, message="Response could not be interpreted from url {}".format(url))
 
-    # TODO: Add JSON-LD COMPACTION
     actions = [add_task(JSONLD_COMPACT_DATA, data=data, node_id=url)]
     return task_result(message="Successfully fetched JSON data from {}".format(url), actions=actions)
 
 
 def jsonld_compact_data(state, task_meta):
     # TODO: Cache-friendly JSON-LD compaction into the Open Badges context
-    pass
+    input_data = json.loads(task_meta.get('data'))
```

Comment:
`task_meta.get('data', '{}')` perhaps. That would avoid blowup on the json.loads. I'm probably actually going to go with a try/except TypeError here so I can return a relevant error message in the task_result instead of letting it go through to the rest of the task that queues up other actions on it.